ecl/eclcmd/eclcmd_common.cpp

```diff
@@ -895,3 +896,38 @@ eclCmdOptionMatchIndicator EclCmdWithQueryTarget::parseCommandLineOptions(ArgvIt
     }
     return EclCmdOptionMatch;
 }
+
+void EclCmdOptionsDFU::preallocatePublisherWuid(EclCmdCommon &cmd)
+{
+    Owned<IClientFileSpray> client = createCmdClientExt(FileSpray, cmd, "");
+    Owned<IClientCreateDFUPublisherWorkunit> req = client->createCreateDFUPublisherWorkunitRequest();
+    Owned<IClientCreateDFUPublisherWorkunitResponse> resp;
+    try
+    {
+         resp.setown(client->CreateDFUPublisherWorkunit(req));
+    }
+    catch (IException *E)
+    {
+        //if we can't preallocate the publisher DFU workunit it might be because the server is too old, an expected scenario pre/early release 9.x, we'll just create the publisher wuid the old way
+        StringBuffer msg;
+        int code = E->errorCode();
+        E->errorMessage(msg);
+        E->Release();
+
+        //Not ideal way of detecting this, but all we're trying to do is to improve the info message when the server is old
+        if (code == -2 && strieq(msg, "400: Bad Request [Method CreateDFUPublisherWorkunit not available in service FileSpray"))
```

Comment:
This looks like it is missing a closing ']', but that may be correct.  Looking at the code I suspect not.

Possibly better would be

strstr(msg, "Method CreateDFUPublisherWorkunit not available")

Reply:
The string is correct, the closing ']' is missing I did test this.

But agreed, I'll switch to a strstr.